airbyte-cdk/bulk/toolkits/load-low-code/src/main/kotlin/io/airbyte/cdk/load/discoverer/Field.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.discoverer
+
+import com.fasterxml.jackson.databind.JsonNode
+import io.airbyte.cdk.load.data.FieldType
+import java.util.function.Predicate
+
+/** Potential improvement: memoize at least isMatchingKey as it will be called multiple times */
+class Field(
+    private val apiRepresentation: JsonNode,
+    namePath: List<String>,
+    private val typePath: List<String>,
+    matchingKeyPredicate: Predicate<JsonNode>,
+    availabilityPredicate: Predicate<JsonNode>,
+    requiredPredicate: Predicate<JsonNode>,
+    private val typeMapper: Map<String, FieldType>,
```

Comment:
This is interesting as the FieldType also includes nullability. We probably need to update this to return a `AirbyteType` and eventually, we can add `Predicate<JsonNode>` to define nullability. It is probably not really relevant right now because [nullability has been reverted](https://github.com/airbytehq/airbyte/pull/62854) and there are no plans to move forward with it but we should probably still change the type to `AirbyteType` here